net-p2p/gridcoin/gridcoin-5.3.3.0.ebuild

```diff
@@ -0,0 +1,153 @@
+# Copyright 2022 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+inherit autotools flag-o-matic desktop systemd
+
+DESCRIPTION="Gridcoin Proof-of-Stake based crypto-currency that rewards BOINC computation"
+HOMEPAGE="https://gridcoin.us/"
+
+if [[ ${PV} == 9999 ]]; then
+	inherit git-r3
+	EGIT_REPO_URI="https://github.com/gridcoin/Gridcoin-Research.git"
+	EGIT_BRANCH="development"
+	PATCHES="${FILESDIR}/gridcoin-9999-desktop.patch"
+else
+	SRC_URI="https://github.com/gridcoin/Gridcoin-Research/archive/refs/tags/${PV}.tar.gz -> ${P}.tar.gz"
+	KEYWORDS="~amd64"
+	S="${WORKDIR}/Gridcoin-Research-${PV}"
+fi
+
+SLOT="0"
+LICENSE="MIT"
+
+IUSE_GUI="dbus gui"
+IUSE_DAEMON="daemon"
+IUSE_OPTIONAL="bench +boinc debug +hardened libraries pic qrcode static test upnp utils"
+IUSE="${IUSE_GUI} ${IUSE_DAEMON} ${IUSE_OPTIONAL}"
+
+RESTRICT="!test? ( test )"
+
+REQUIRED_USE="|| ( daemon gui ) dbus? ( gui ) qrcode? ( gui )"
+
+RDEPEND="
+	acct-group/gridcoin
+	acct-user/gridcoin
+	>=dev-libs/boost-1.73.0:=
+	dev-libs/libevent
+	>=dev-libs/libzip-1.3.0:=
+	>=dev-libs/openssl-1.1.1d:=
+	net-misc/curl
+	sys-libs/db:5.3[cxx]
+	boinc? ( sci-misc/boinc )
+	dbus? ( dev-qt/qtdbus:5 )
+	gui? (
+		dev-qt/linguist-tools:5 \
+		dev-qt/qtcharts:5 \
+		dev-qt/qtconcurrent:5 \
+		dev-qt/qtcore:5 \
+		dev-qt/qtgui:5 \
+		dev-qt/qtnetwork:5 \
+		dev-qt/qtwidgets:5 \
+	)
+	qrcode? ( media-gfx/qrencode )
+	upnp? ( >=net-libs/miniupnpc-1.9.20140401 )
+	utils? ( net-p2p/bitcoin-cli dev-util/bitcoin-tx )"
+DEPEND="${RDEPEND}"
+
+pkg_setup() {
+	BDB_VER="$(best_version sys-libs/db:5.3)"
+	export BDB_CFLAGS="-I${ESYSROOT}/usr/include/db${BDB_VER:12:3}"
+	export BDB_LIBS="-ldb_cxx-${BDB_VER:12:3}"
+}
+
+src_prepare() {
+	default
+	eautoreconf
+}
+
+src_configure() {
+	append-flags -Wa,--noexecstack
+	econf \
+		$(use_enable bench) \
+		$(use_enable debug) \
+		$(use_enable hardened) \
+		$(use_enable static) \
+		$(use_enable test tests) \
+		$(use_with daemon) \
+		$(use_with dbus qtdbus) \
+		$(use_with gui gui qt5) \
+		$(use_with libraries libs) \
+		$(use_with pic) \
+		$(use_with qrcode qrencode) \
+		$(use_with upnp miniupnpc) \
+		$(use_with utils)
+}
+
+src_install() {
+	# Live package is called staging and should *only* be used for testing purposes
+	if [[ ${PV} == 9999 ]]
```

Comment:
Let's use consistent if style, I prefer the style used in pkg_postinst but if you want to use this style instead, that's fine, just change the other instances.

also, nit: I'd local suffix either way and just set the value conditionally.